src/common/transactions/transaction-utils.ts

```diff
@@ -119,3 +146,119 @@ export const generateSTXTransferTx = ({
+    const date = time ? isoDateToLocalDateSafe(time) : undefined;
+    if (date && txHasTime(tx)) {
+      if (!txsByDate.has(date)) {
+        txsByDate.set(date, []);
+      }
+      txsByDate.set(date, [...(txsByDate.get(date) || []), atx]);
+    }
+    if (!txHasTime(tx)) {
+      const today = todaysIsoDate();
+      txsByDate.set(today, [...(txsByDate.get(today) || []), atx]);
+    }
+    return txsByDate;
+  }, new Map<string, AddressTransactionWithTransfers[]>());
+}
+
+function formatTxDateMapAsList(txMap: Map<string, AddressTransactionWithTransfers[]>) {
+  return [...txMap.keys()].map(date => ({
+    date,
+    displayDate: displayDate(date),
+    txs: txMap.get(date) ?? [],
+  }));
+}
+
+function filterDuplicateTx(txs: AddressTransactionWithTransfers[]) {
+  return txs.filter(atx => {
+    const countOfCurrentTxid = txs.filter(({ tx: { tx_id } }) => atx.tx.tx_id === tx_id).length;
+    const isDropped = atx.tx.tx_status.includes('dropped');
+    if (countOfCurrentTxid === 1 && !isDropped) return true;
+    return atx.tx.tx_status === 'success' || atx.tx.tx_status.includes('abort');
+  });
+}
+
+export function createTxDateFormatList(txs: AddressTransactionWithTransfers[]) {
+  return formatTxDateMapAsList(groupTxsByDateMap(filterDuplicateTx(txs)));
+}
+
+export const getTxCaption = (transaction: Tx) => {
+  if (!transaction) return null;
+  switch (transaction.tx_type) {
+    case 'smart_contract':
+      return truncateMiddle(transaction.smart_contract.contract_id, 4);
+    case 'contract_call':
+      return transaction.contract_call.contract_id.split('.')[1];
+    case 'token_transfer':
+    case 'coinbase':
+    case 'poison_microblock':
+      return truncateMiddle(transaction.tx_id, 4);
+    default:
+      return null;
+  }
+};
+
+const getAssetTransfer = (tx: Tx): TransactionEventFungibleAsset | null => {
+  if (tx.tx_type !== 'contract_call') return null;
+  if (tx.tx_status !== 'success') return null;
+  const transfer = tx.events.find(event => event.event_type === 'fungible_token_asset');
+  if (transfer?.event_type !== 'fungible_token_asset') return null;
+  return transfer;
+};
+
+export const getTxValue = (tx: Tx, isOriginator: boolean): number | string | null => {
+  if (tx.tx_type === 'token_transfer') {
+    return `${isOriginator ? '-' : ''}${stacksValue({
+      value: tx.token_transfer.amount,
+      withTicker: false,
+    })}`;
+  }
+  const transfer = getAssetTransfer(tx);
+  if (transfer) return new BigNumber(transfer.asset.amount).toFormat();
+  return null;
+};
+
+export const getTxTitle = (tx: Tx) => {
+  switch (tx.tx_type) {
+    case 'token_transfer':
+      return 'Stacks Token';
+    case 'contract_call':
+      return tx.contract_call.function_name;
+    case 'smart_contract':
+      return getContractName(tx.smart_contract.contract_id);
+    case 'coinbase':
+      return `Coinbase ${(tx as CoinbaseTransaction).block_height}`;
```

Comment:
Is this type cast necessary? The types use discriminating unions, so here it should know it's a coinbase tx type. If not there is something wrong.

Reply:
I just moved this function in this file, not sure why that why there. But if I remove it does not type.